### PR TITLE
Return full file paths on Electron.

### PR DESCRIPTION
**Pull request by wali-s**
The file paths were being overwritten because that property doesn't exist on web but it does on electron.

The existing behaviour is identical on web and electron.

This PR will keep things unchanged on web and make the `path` property an absolute path on electron.

An alternative solution would be to rename the `path` that `react-dropzone` uses so it doesn't conflict.

Ref: https://github.com/react-dropzone/react-dropzone/issues/804

### CODE REVIEW

File: src/file.ts [hunk 1]
```diff
@@ -15,22 +15,25 @@ export const COMMON_MIME_TYPES = new Map([
 ]);
 
 
-export function toFileWithPath(file: File, path?: string): FileWithPath {
+export function ensureFileWithPath(file: FileWithPath, path?: string): FileWithPath {
```
Comment (rolandjitsu) [inline src/file.ts]: I don't see the point in renaming this fn 😕 It just adds unnecessary changes. Please undo,

File: src/file.ts [hunk 2]
```diff
@@ -15,22 +15,25 @@ export const COMMON_MIME_TYPES = new Map([
...
-            // the File will have a {webkitRelativePath} property
-            // https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/webkitdirectory
-            : typeof webkitRelativePath === 'string' && webkitRelativePath.length > 0
-                ? webkitRelativePath
-                : file.name,
-        writable: false,
-        configurable: false,
-        enumerable: true
-    });
+    if (!('path' in f)) { // on electron, path is already set to the absolute path
```
Comment (rolandjitsu) [inline src/file.ts]: Check for `typeof f.path !== 'string'` instead.

**Comment by wali-s** [comment]
Thanks for the review! Changes made 😄
